pkg/driver/controller.go

```diff
@@ -502,6 +505,18 @@ func (cs *controller) DeleteVolume(
 	unlock := cs.volumeLock.LockVolume(volumeID)
 	defer unlock()
 
+	// Fetch the list of snapshot for the given volume
```

Comment:
The below code block should ideally be placed before line 542.

Reply:
Here I am declaring a new variable `snapList` , if i put it at line 542 , it error out stating `goto deleteResponse jumps over variable declaration`. 

`deleteResponse` is used to check if the volume is already deleted.

Reply:
Seems to be a Go limitation. Not sure if it's addressed with this issue: https://github.com/golang/go/issues/27165. There are ways around this, eg. declare the variable on the top.

The current logic seems incorrect.

Reply:
Why do we need to declare it at the top?
Can't we just do snapList, err := ..... ?

Reply:
no, there is a `goto deleteResponse` which checks if volume is present . If we do `:=` then its errors saying `goto deleteResponse jumps over variable declaration.`

Reply:
Let's not use `goto`, it's generally seen as a bad practice nowadays.

Reply:
removed `goto`